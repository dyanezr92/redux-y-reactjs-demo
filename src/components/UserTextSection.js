import React from 'react'
import PropTypes from 'prop-types'
import UserTextList from './UserTextList'
import UserTextForm from './UserTextForm'
import Col from './Col'

const UserTextSection = ({ elements, elementName, onDeleteClick, onSubmit }) => {

  return (
    <Col w={5} style={{ display: "inline-block", margin: 20 }}>
      <UserTextList elements={elements} elementName={elementName} onDeleteClick={onDeleteClick}/>
      <UserTextForm onUserTextSubmit={ onSubmit } elementName={ elementName } />
    </Col>
  );
};

UserTextSection.propTypes = {
  elements: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string,
    author: PropTypes.string
  }).isRequired).isRequired,
  elementName: PropTypes.string.isRequired,
  onDeleteClick: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
}

export default UserTextSection
