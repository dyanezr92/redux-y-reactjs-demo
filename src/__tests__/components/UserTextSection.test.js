import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store'; // Smart components

import UserTextSection from '../../components/UserTextSection';

describe('<UserTextSection />', () => {
  describe('render()', () => {
    test('renders the user text section', () => {
      const wrapper = shallow(
        <UserTextSection elements={[ {"id": 1, "author": "Finn", "text": "Awesome comment!"}]}
          elementName="comments" onDeleteClick={ console.log } onSubmit={ console.log } />
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
