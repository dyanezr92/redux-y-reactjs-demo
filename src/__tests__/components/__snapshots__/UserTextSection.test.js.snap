// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserTextSection /> render() renders the user text section 1`] = `
<Col
  style={
    Object {
      "display": "inline-block",
      "margin": 20,
    }
  }
  w={5}
>
  <UserTextList
    elementName="comments"
    elements={
      Array [
        Object {
          "author": "Finn",
          "id": 1,
          "text": "Awesome comment!",
        },
      ]
    }
    onDeleteClick={[Function]}
  />
  <UserTextForm
    elementName="comments"
    onUserTextSubmit={[Function]}
  />
</Col>
`;
